libraries-3/src/main/java/com/baeldung/mail/attachment/EmailAttachmentReceiver.java

```diff
@@ -0,0 +1,108 @@
+        this.userName = userName;
+        this.password = password;
+    }
+
+    public void downloadEmailAttachmentsTo(String directory) {
+        Properties properties = new Properties();
+
+        // server settings
+        properties.put(String.format(MAIL_PROTOCOL_HOST, this.protocol), this.host);
+        properties.put(String.format(MAIL_PROTOCOL_PORT, this.protocol), this.port);
+
+        // SSL settings
+        properties.put(String.format(MAIL_PROTOCOL_SOCKET_FACTORY_CLASS, this.protocol), JAVAX_NET_SSL_SSL_SOCKET_FACTORY);
+        properties.put(String.format(MAIL_PROTOCOL_SOCKET_FACTORY_FALLBACK, this.protocol), FALSE);
+        properties.put(String.format(MAIL_PROTOCOL_SOCKET_FACTORY_PORT, this.protocol), this.port);
+
+        // obtaining sessions
+        Session session = Session.getDefaultInstance(properties);
+        try {
+                // connecting to the mail server
+                Store store = session.getStore(this.protocol);
+                store.connect(this.userName, this.password);
+
+            // opens the inbox folder
+            Folder inbox = store.getFolder(INBOX);
+            inbox.open(Folder.READ_ONLY);
+            List<Message> messages = asList(inbox.getMessages());
+            for (Message message : messages) {
+                // filtering message with content type 'multipart'
+                if (message.getContentType()
+                    .equalsIgnoreCase(MULTIPART)) {
+                    checkForAttachmentsAndSave(message, directory);
+                }
+            }
+            // closing open connections
+            inbox.close();
+            store.close();
+        } catch (NoSuchProviderException noSuchProviderException) {
+            LOGGER.error("An error occurred while downloading attachments", noSuchProviderException);
+        } catch (MessagingException messagingException) {
+            LOGGER.error("An error occurred while downloading attachments", messagingException);
+        } catch(IOException ioException) {
+            LOGGER.error("An error occurred while downloading attachments", ioException);
+        }
+    }
+
+    private void checkForAttachmentsAndSave(Message message, String directory) throws IOException, MessagingException {
+        Multipart multipart = (Multipart) message.getContent();
+        for (int partCount = 0; partCount < multipart.getCount(); partCount++) {
+            if (isAttachment(partCount, multipart)) {
+                // this is the attachment.
+                MimeBodyPart mimeBodyPart = (MimeBodyPart) multipart.getBodyPart(partCount);
+                mimeBodyPart.saveFile(directory + File.separator + mimeBodyPart.getFileName());
+            }
+        }
+    }
+
+    private boolean isAttachment(int partCount, Multipart multipart) throws MessagingException {
+        MimeBodyPart mimeBodyPart = (MimeBodyPart) multipart.getBodyPart(partCount);
+        return Part.ATTACHMENT.equalsIgnoreCase(mimeBodyPart.getDisposition());
```

Comment:
When we use `equalsIgnoreCase`, it has to be possible for mixed case answers to be valid. Is that the case here?